v21.2/build-a-nodejs-app-with-cockroachdb-prisma.md

```diff
@@ -0,0 +1,167 @@
+~~~
+
+The `index.js` file contains the code for `INSERT`, `SELECT`, `UPDATE`, and `DELETE` SQL operations:
+
+{% include_cached copy-clipboard.html %}
+~~~ js
+{% remote_include https://raw.githubusercontent.com/cockroachlabs/example-app-node-prisma/main/index.js %}
+~~~
+
+All of the database operations are wrapped in a helper function named `retryTxn`. This function attempts to commit statements in the context of an explicit transaction. If a [retry error](transaction-retry-error-reference.html) is thrown, the wrapper will retry committing the transaction, with [exponential backoff](https://en.wikipedia.org/wiki/Exponential_backoff), until the maximum number of retries is reached (by default, 15).
+
+## Step 3. Initialize the database
+
+{{site.data.alerts.callout_info}}
+CockroachDB does not support Prisma Migrate. We recommend executing DDL SQL statements to initialize your database schema separately. After the database schema is initialized, you can load existing tables into your Prisma schema with the Prisma Client, as demonstrated [here](https://www.prisma.io/docs/concepts/components/introspection).
+{{site.data.alerts.end}}
+
+To initialize the example database schema, use the [`cockroach sql`](cockroach-sql.html) command to execute the SQL statements in the `dbinit.sql` file:
+
+<div class="filter-content" markdown="1" data-scope="cockroachcloud">
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+cat dbinit.sql | cockroach sql --url "<connection-string>"
+~~~
+
+Where `<connection-string>` is the connection string you obtained earlier from the {{ site.data.products.db }} Console.
+
+</div>
+
+<div class="filter-content" markdown="1" data-scope="local">
+
+{% include_cached copy-clipboard.html %}
+~~~ shell
+cat dbinit.sql | cockroach sql --url "postgresql://root@localhost:26257?sslmode=disable"
+~~~
+
+{{site.data.alerts.callout_info}}
+`postgresql://root@localhost:26257?sslmode=disable` is the `sql` connection string you obtained earlier from the `cockroach` welcome text.
+{{site.data.alerts.end}}
+
+</div>
+
+## Step 4. Configure the project
+
+1. Update the `DATABASE_URL` environment variable in the project's `.env` file with a valid connection string to your cluster:
+
+    <div class="filter-content" markdown="1" data-scope="cockroachcloud">
+
+    {% include_cached copy-clipboard.html %}
+    ~~~
+    DATABASE_URL="<connection-string>"
+    ~~~
+
+    Where `<connection-string>` is the connection string you obtained earlier from the {{ site.data.products.db }} Console.
+
+    </div>
+
+    <div class="filter-content" markdown="1" data-scope="local">
+
+    {% include_cached copy-clipboard.html %}
+    ~~~
+    DATABASE_URL="postgresql://root@localhost:26257?sslmode=disable"
+    ~~~
+
+    </div>
+
+1. Install the app requirements:
+
+    {% include_cached copy-clipboard.html %}
+    ~~~ shell
+    $ npm install prisma uuid --save-dev @prisma/client
```

Comment:
It should be possible to install the requirements with just `npm install` as long as they're specified in `package.json`.